parsers/ES.py

```diff
@@ -163,40 +243,46 @@ def fetch_production(
 ) -> List[dict]:
     check_valid_parameters(zone_key, session, target_datetime)
     ses = session or Session()
-    island_data = fetch_island_data(zone_key, ses, target_datetime)
+    fuel_mapping = FUEL_MAPPING.copy()
+
+    ## Production mapping override for Canary Islands
+    # NOTE the LNG terminals are not built yet, so power generated by "gas" or "cc" in ES-CN domain is actually using oil.
+    # Recheck this every 6 months and move to gas key if there has been a change.
+    # Last checked: 2022-06-27
+    if zone_key.split("-")[1] == "CN":
+        fuel_mapping["gas"] = "oil"
+        fuel_mapping["cc"] = "oil"
+
+    if zone_key == "ES-IB-ME" or zone_key == "ES-IB-FO":
+        fuel_mapping["gas"] = "oil"
+    if zone_key == "ES-IB-IZ":
+        fuel_mapping["die"] = "gas"
+
+    island_data = fetch_island_data(zone_key, ses, target_datetime, fuel_mapping)
     productionEventList = ProductionBreakdownList(logger)
-
     for event in island_data:
         production = ProductionMix()
+        for mode in [
+            "biomass",
+            "coal",
+            "gas",
+            "geothermal",
+            "nuclear",
+            "oil",
+            "solar",
+            "wind",
+            "unknown",
+        ]:
```

Comment:
It should loop over all modes here, not just the ones that has a defined mode so we can log if something is missing.